Add OPTIONS method to POST endpoint

nshumoogum opened the pull request:
### What

See title

### How to review

Check for any typo

### Who can review

Anyone

nshumoogum committed:
Fix scripts

nshumoogum committed:
Add options method to POST request endpoint

dandwal reviewed (commented):
I think we need to add the following headers and return a status code of 204 is it is a preflight check.

Frontend / browser will send something like:
Access-Control-Request-Method: OPTIONS 
Access-Control-Request-Headers: origin
Origin: https://somelovelyplace.com

And backend needs to return:

HTTP/1.1 204 No Content
Connection: keep-alive
Access-Control-Allow-Origin: * // I think * (wildcard) is best as we may use it in other POCs
Access-Control-Allow-Methods: POST, OPTIONS
Access-Control-Max-Age: 86400 // or whatever value

Otherwise the browser will throw a tantrum and block the real POST request

nshumoogum committed:
Fix parent search validation of request data

Validation model handles multi polygons and polygon shapes

Also added necessary headers for OPTIONS request method

nshumoogum committed:
Add OPTIONS method to all endpoints

dandwal reviewed (approved):
Haven't tested pining this from a browser but from the look of it everything seems to be there

nshumoogum committed:
Add OPTIONS method to /search/parent endpoint
